pkg/api/types.go

```diff
@@ -1824,6 +1824,9 @@ type PodSpec struct {
 	// If not specified, the pod will not have a domainname at all.
 	// +optional
 	Subdomain string `json:"subdomain,omitempty"`
+	// If specified, the pod's scheduling preferences
+	// +optional
```

Comment:
omitempty and +optional, and if you need to support empty vs nil arrays or maps you should not set omitempty and instead use the same construct that ExtraValues in authorization uses.